lib/fs/fsfileinfo_windows.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2017 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package fs
+
+import (
+	"os"
+)
+
+func (e fsFileInfo) Mode() FileMode {
+	m := e.FileInfo.Mode()
+	if m&os.ModeSymlink != 0 && e.Size() > 0 {
+		// "Symlinks" with nonzero size are in fact "hard" links, such as
+		// NTFS deduped files. Remove the symlink bit.
+		m &^= os.ModeSymlink
+	}
+	return FileMode(m)
+}
+
+func (e fsFileInfo) IsRegular() bool {
+	if m&os.ModeSymlink != 0 && e.Size() > 0 {
+		// Take out the symlink bit and see if it's a regular *now*... This
+		// is a bit of extra carefulness against things that might have the
+		// symlink bit set, have a size, yet have some other bit set that
+		// makes them non-regular.
+		m := e.FileInfo.Mode() &^ os.ModeSymlink
+		return m.IsRegular()
+	}
+	return e.FileInfo.Mode().IsRegular()
+}
+
+func (e fsFileInfo) IsSymlink() bool {
+	// N.B. this uses *our* Mode() which appropriately masks this bit.
+	return e.Mode()&FileMode(os.ModeSymlink) != 0
```

Comment:
Just ModeSymlink, which is already defined in fs.go.

Reply:
It actually isn't but maybe it should be